### PR TITLE
Feature/interleaved transform feedback

### PR DESCRIPTION
**Pull request by darionco**
Added `transformFeedbackMode` option to `Program` creation.
Useful to choose between SEPARATE_ATTRIBS and INTERLEAVED_ATTRIBS. Can be used for computing interleaved attribute buffers and other GPGPU tasks that might require multiple outputs from a single input.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -638,12 +638,13 @@ export class App {
         @param {Object} [options] Texture options.
         @param {Object} [options.attributeLocations] Map of attribute names to locations (useful when using GLSL 1).
         @param {Array} [options.transformFeedbackVaryings] Array of varying names used for transform feedback output.
+        @param {GLenum} [options.transformFeedbackMode] Capture mode of the transform feedback. (Default: PicoGL.SEPARATE_ATTRIBS).
         @return {Program} New Program object.
     */
     createProgram(vsSource, fsSource, opts = {}) {
-        let {transformFeedbackVaryings, attributeLocations} = opts;
+        let {transformFeedbackVaryings, attributeLocations, transformFeedbackMode} = opts;
```
Comment (tsherif) [inline src/app.js:645]: Make sure `App.createPrograms` supports the new option as well.

**Comment by tsherif** [comment]
Nice! Would you be up for writing a test for this? I don't like having functionality that isn't used in any examples or tests. I'm thinking an easy one would be to copy the particles example into `tests/render` (maybe call it `particles-interleaved`) and have it use the new interleave mode.

**Comment by darionco** [comment]
Yup, I'll work it out, for now it seems like I am going to add a few more features to this PR, like reading data back from the transform feedback buffer for GPGPU computations. I'll add tests and examples once I am done, thanks!

**Comment by tsherif** [comment]
Oooh, that would be awesome! One thing is that I'd love for readbacks to be async (https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API/WebGL_best_practices#Non-blocking_async_data_downloadreadback). I was thinking of doing something like this for `readPixels`.

**Comment by darionco** [comment]
Sorry for the long wait.

I have added a very simple example for using interleaved transform feedback buffers. Let me know if that is enough and I'll add a test for it.

As for the GPGPU features I had been planning, I won't be adding them to PicoGL, over the holidays I researched and tested WebGPU and decided to use it since it provides a much more powerful API and it includes a native compute pipeline. I do see the need for a PicoGL-like library for WebGPU though, there's a lot of boilerplate code needed to do simple operations. Let me know if you would like to collaborate on something like that. 

**unrelated**
During my research I realized that I find you everywhere I look that has to do with web graphics, in fact I just noticed that you contributed to luma.gl, I used to like luma.gl but it grew to be exactly what I was trying to avoid: a bunch of layers of abstraction with a ton of bloat that I am probably never going to use. Anyway, I remember that back when I started using it the documentation linked to PicoGL's order independent transparency example, that's how I ended up replacing it with PicoGL!

I also explored your WebGPU experiments, really nice, did you find any red flags that I should know about? (other than being an early adopter, that's not a problem for my work luckily)

Anyway, thanks for your awesome work!

**Comment by tsherif** [comment]
Great, thanks!

That's funny :) I have been around the WebGL world for a while now. I was tech lead on luma.gl while at Uber, and before that I worked heavily on SceneJS while I was at BioDigital.

WebGPU will definitely make the GPGPU stuff easier. I can't remember any specific gotchas, though I remember moving data around was a little trickier, like that you can't upload data directly to a texture, it has to go to a buffer first. Not sure if that's changed , it's been a while since I've looked at the spec. In general, my feeling was that the API is a bit saner than WebGL, in particular since most state is maintained in pipelines rather than globally.

**Pull request merged**